src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -752,13 +763,135 @@ public void shouldReuseMainConnection() throws SQLException {
                 new RouteDecisionBuilder(Reason.MAIN_CONNECTION_REUSE)
                     .sql(SIMPLE_QUERY)
                     .cause(
-                        new RouteDecisionBuilder(Reason.RW_API_CALL).sql(SIMPLE_QUERY).build()
+                        new RouteDecisionBuilder(Reason.RW_API_CALL).sql(SIMPLE_QUERY).isSqlWrite(true).build()
                     )
+                    .isSqlWrite(false) //read which was treated as write
                     .build()
             )
         );
     }
 
+    @Test
+    public void shouldShowThatMainConnectionCantBeReusedAsRead() throws SQLException {
```

Comment:
It seems the test abuses Mockito. Can you rewrite it without (or at least limit the usage) Mockito to improve readability?

Reply:
It seems like the test tests multiple things at once can it follow given-when-then or arrange-act-assert pattern?

Reply:
done